src/typing_extensions.py

```diff
@@ -97,7 +97,7 @@ def _check_generic(cls, parameters, elen=_marker):
     alen = len(parameters)
     if alen != elen:
         if hasattr(cls, "__parameters__"):
-            parameters = [p for p in cls.__parameters__ if not _is_unpack(p)]
+            parameters = (p for p in cls.__parameters__ if not _is_unpack(p))
```

Comment:
Are you sure this is faster? As @hauntsaninja has pointed out elsewhere, list comprehensions are often faster than generator expressions if the list is small.

Reply:
You're right. 
I will suggest another change: replace two iterables with one by inlining one into another:
```py
parameters = [p for p in cls.__parameters__ if not _is_unpack(p)]
num_tv_tuples = sum(isinstance(p, TypeVarTuple) for p in parameters)
# replace with:
num_tv_tuples = sum(isinstance(parameter, TypeVarTuple) for parameter in cls.__parameters__ if not _is_unpack(parameter))
```
Inlining gives us 10-20% speed-up.

Here are the benchmarks:
```py
# iterating over list
$>py -m timeit -s "pars=['a','b','cc','d','ee','ff'];cond=lambda x:len(x)%2==0" "it=[p for p in pars if cond(p)];sum(isinstance(p,str) for p in it)"
200000 loops, best of 5: 1.18 usec per loop

# iterating over generator
$>py -m timeit -s "pars=['a','b','cc','d','ee','ff'];cond=lambda x:len(x)%2==0" "it=(p for p in pars if cond(p));sum(isinstance(p,str) for p in it)"
200000 loops, best of 5: 1.26 usec per loop

# inline first generator into second
$>py -m timeit -s "pars=['a','b','cc','d','ee','ff'];cond=lambda x:len(x)%2==0" "sum(isinstance(p,str) for p in pars if cond(p))"
500000 loops, best of 5: 984 nsec per loop

$>py -m timeit -s "pars=[*map(str,range(10000))];cond=lambda x:len(x)%2==0" "it=[p for p in pars if cond(p)];sum(isinstance(p,str) for p in it)"
200 loops, best of 5: 1.52 msec per loop

$>py -m timeit -s "pars=[*map(str,range(10000))];cond=lambda x:len(x)%2==0" "it=(p for p in pars if cond(p));sum(isinstance(p,str) for p in it)"
200 loops, best of 5: 1.71 msec per loop

$>py -m timeit -s "pars=[*map(str,range(10000))];cond=lambda x:len(x)%2==0" "sum(isinstance(p,str) for p in pars if cond(p))"
200 loops, best of 5: 1.38 msec per loop
```